tests/tests_qdevice_tls_kaptb_options.yml

```diff
@@ -13,6 +13,24 @@
             (ansible_play_hosts_all | length == 1)
             | ternary('localhost', ansible_play_hosts[0]) }}"
 
+    - name: Ensure correct package manager for ostree systems
+      vars:
+        ostree_pkg_mgr: ansible.posix.rhel_rpm_ostree
+        ostree_booted_file: /run/ostree-booted
+      when: ansible_facts.pkg_mgr | d("") != ostree_pkg_mgr
+      block:
+        - name: Check if system is ostree
+          stat:
+            path: "{{ ostree_booted_file }}"
+          register: __ostree_booted_stat
+
+        - name: Set package manager to use for ostree
+          ansible.utils.update_fact:
+            updates:
+              - path: ansible_facts.pkg_mgr
+                value: "{{ ostree_pkg_mgr }}"
+          when: __ostree_booted_stat.stat.exists
+
```

Comment:
Just out of curiosity, why is this qdevice test allowed on ostree and tests_qdevice_all_options and tests_qdevice_minimal is not?

Reply:
those tests call (eventually) test_cleanup_qnetd.yml which tries to ensure the package `corosync-qnetd` is removed.  Packages cannot be removed from an rpm-ostree system.  Maybe the test can be run if we just skip the task `Make sure qnetd is not installed` on rpm-ostree systems?  I'm not sure why the test is removing that package.

Reply:
This test runs the cleanup as well. It just needs new pcs (so new it hasn't been released yet) to run, otherwise the test exits without testing anything.

If I remember correctly, the reason for removing the package is to test that the role installs it.

Reply:
> If I remember correctly, the reason for removing the package is to test that the role installs it.

ok - then I will change my PR to skip only that task and not the entire tests

Reply:
I think that the reason for installing and uninstalling qnetd is to maintain a defined environment for the case when all tests are run on the same node.

Reply:
hmm - so I changed it so that all of the tests are run, and the test that removes the package is skipped.  Now I get this error:

```
TASK [linux-system-roles.ha_cluster : Set up qnetd] ****************************
task path: /home/rmeggins/linux-system-roles/ha_cluster/tests/roles/linux-system-roles.ha_cluster/tasks/test_setup_qnetd.yml:16
Thursday 09 November 2023  11:09:52 -0700 (0:00:00.377)       0:00:03.178 *****
fatal: [/home/rmeggins/.cache/linux-system-roles/rhel-8-10-ostree.qcow2]: FAILED! => {
...
STDOUT:

Quorum device 'net' initialized
Starting quorum device...


STDERR:

Error: Unable to start corosync-qnetd: Job for corosync-qnetd.service failed because the control process exited with error code.
See "systemctl status corosync-qnetd.service" and "journalctl -xe" for details.
```

This is from the journal

```
Nov 09 18:09:53 ibm-p8-kvm-03-guest-02.virt.pnr.lab.eng.rdu2.redhat.com corosync-qnetd[934]: Can't open NSS DB directory (13): Permission denied
```

Is the NSS DB in /usr somewhere?

Reply:
The NSS DB is in /etc/corosync/qnetd/ for qnetd, and in /etc/corosync/qdevice/net/ for qdevice.

Reply:
@tomjelinek This is what those files/directories look like - do they have the correct permissions/ownership?

```
/etc/corosync/qdevice/net/:
total 8
drwxr-xr-x. 3 root root 4096 Nov 10 14:12 ../
drwxrwx---. 2 root root 4096 Nov 10 14:12 ./

/etc/corosync/qnetd/:
total 12
drwxr-xr-x. 5 root root 4096 Nov 10 14:17 ../
drwxr-xr-x. 3 root root 4096 Nov 10 14:17 ./
drwxrwx---. 2 root root 4096 Nov 10 14:17 nssdb/

/etc/corosync/qnetd/nssdb:
total 108
drwxr-xr-x. 3 root root  4096 Nov 10 14:17 ../
-rw-rw----. 1 root root     0 Nov 10 14:17 pwdfile.txt
-rw-rw----. 1 root root   432 Nov 10 14:17 pkcs11.txt
-rw-rw----. 1 root root    41 Nov 10 14:17 noise.txt
-rw-r--r--. 1 root root  4272 Nov 10 14:17 qnetd-cacert.crt
-rw-rw----. 1 root root     4 Nov 10 14:17 serial.txt
-rw-rw----. 1 root root 53248 Nov 10 14:17 key4.db
-rw-rw----. 1 root root 28672 Nov 10 14:17 cert9.db
drwxrwx---. 2 root root  4096 Nov 10 14:17 ./
```
basically, everything is owner/group `root`, and `0640` or `0660`

Reply:
@richm This is what it should look like:
```
# ls -la /etc/corosync/qnetd
total 12
drwxrwx---. 3 coroqnetd coroqnetd 4096 Nov 10 15:54 .
drwxr-xr-x. 6 root      root      4096 Nov 10 15:53 ..
drwxrwx---. 2 coroqnetd coroqnetd 4096 Nov 10 15:54 nssdb

# ls -la /etc/corosync/qnetd/nssdb/
total 108
drwxrwx---. 2 coroqnetd coroqnetd  4096 Nov 10 15:54 .
drwxrwx---. 3 coroqnetd coroqnetd  4096 Nov 10 15:54 ..
-rw-rw----. 1 coroqnetd coroqnetd 28672 Nov 10 15:54 cert9.db
-rw-rw----. 1 coroqnetd coroqnetd 53248 Nov 10 15:54 key4.db
-rw-rw----. 1 coroqnetd coroqnetd    41 Nov 10 15:54 noise.txt
-rw-rw----. 1 coroqnetd coroqnetd   432 Nov 10 15:54 pkcs11.txt
-rw-rw----. 1 coroqnetd coroqnetd     0 Nov 10 15:54 pwdfile.txt
-rw-r--r--. 1 coroqnetd coroqnetd  4272 Nov 10 15:54 qnetd-cacert.crt
-rw-rw----. 1 coroqnetd coroqnetd     4 Nov 10 15:54 serial.txt
```
I just ran the role and it set permissions correctly. Let me check the tests...

Reply:
Must have something to do with `rpm-ostree` systems - it does something strange with user/group ids - where is the `coroqnetd` user/group created?

Reply:
@richm I'm not sure which tasks you disabled for ostree. Anyway, it is crucial that /etc/corosync/qnetd is owned by coroqnetd:coroqnetd. Then all the files have correct permissions. Installing corosync-qnetd package creates /etc/corosync/qnetd with the correct permissions. Can you check what the permissions are in a fresh ostree image before running the role / tests?

Reply:
> where is the `coroqnetd` user/group created?
They are created by installing corosync-qnetd package